subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/dependencymanagement/SamplesManagingTransitiveDependenciesIntegrationTest.groovy

```diff
@@ -151,6 +151,19 @@ class SamplesManagingTransitiveDependenciesIntegrationTest extends AbstractInteg
         assertSingleLib('guava-23.0.jar')
     }
 
+    @UsesSample("userguide/dependencyManagement/managingTransitiveDependencies/importingDependencyConstraintsFromBOM")
+    def "can import dependency versions from a bom"() {
+        executer.inDirectory(sample.dir)
+
+        when:
+        succeeds(COPY_LIBS_TASK_NAME)
+
+        then:
+        def libs = listFilesInBuildLibsDir()
+        libs.size() == 3
+        libs.any { it.name == 'gson-2.8.2.jar' || it.name == 'dom4j-1.6.1.jar' || it.name == 'xml-apis-1.4.01.jar'}
```

Comment:
Maybe use `libs.findAll { ... }.size() == 3` instead.

Reply:
Why do you think that is needed? There should be nothing else except for the jars in the libs folder.

Reply:
I am with @bmuschko on that one. I very much prefer assertions that cannot pass at all if something goes wrong. And my understanding here is that if you end up with 3 libs, but one of them not one of the expected three, this test will keep on passing.

Reply:
Thanks @ljacomet. I misread the comment .I thought this was only about the `libs.size() == 3` line. I will fix it.